docs/concepts/tools/kubectl/object-management-using-commands.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Managing Kubernetes Objects Using Imperative Commands
+---
+
+{% capture overview %}
+Kubernetes objects can quickly be created, updated, and deleted directly using
+imperative commands built into the `kubectl` command-line tool. This document
+explains how those commands are organized and how to use them to manage live objects.
+{% endcapture %}
+
+{% capture body %}
+
+## Trade-offs
+
+The `kubectl` tool supports three kinds of object management:
+
+* Imperative commands
+* Imperative object configuration
+* Declarative object configuration
+
+See [Kubernetes Object Management](/docs/concepts/tools/kubectl/object-management-overview/)
+for a discussion of the advantages and disadvantage of each kind of object management.
+
+## How to create objects
+
+The `kubectl` tool supports verb-driven commands for creating some of the most common
+object types. The commands are named to be recognizable to users unfamiliar with
+the Kubernetes object types.
+
+- `run`: Create a new Deployment object to run Containers in one or more Pods.
+- `expose`: Create a new Service object to load balance traffic across Pods.
+- `autoscale`: Create a new Autoscaler object to automatically horizontally scale a controller, such as a Deployment.
+
+The `kubectl` tool also supports creation commands driven by object type.
+These commands support more object types and are more explicit about
+their intent, but require users to know the type of objects they intend
+to create.
+
+- `create <objecttype> [<subtype>] <instancename>`
+
+Some objects types have subtypes that you can specify in the `create` command.
+For example, the Service object has several subtypes including ClusterIP,
+LoadBalancer, and NodePort. Here's an example that creates a Service with
+subtype NodePort:
+
+```shell
+kubectl create service nodeport <myservicename>
+```
+
+In the preceding example, the `create service nodeport` command is called
+a subcommand of the `create service` command.
+
+You can use the `-h` flag to find the arguments and flags supported by
+a subcommand:
+
+```shell
+kubectl create service nodeport -h
+```
+
+## How to update objects
+
+The `kubectl` command supports verb-driven commands for some common update operations.
+These commands are named to enable users unfamiliar with Kubernetes
+objects to perform updates without knowing the specific fields
+that must be set:
+
+- `scale`: Horizontally scale a controller to add or remove Pods by updating the replica count of the controller.
+- `annotate`: Add or remove an annotation from an object.
+- `label`: Add or remove a label from an object.
+
+The `kubectl` command also supports update commands driven by field name.
+These commands require the user to know the names of the fields
+they want to update:
+
+- `set` <field>: Set the value of a specific field of an object.
```

Comment:
`set` is more nuanced than just "field", it tries to be polymorphic in setting a particular thing in different types of objects. For example "kubectl set selector" sets a selector in different locations depending on the type of resource given to it